src/LibraryViewExtension/Handlers/ResourceHandlerFactory.cs

```diff
@@ -47,8 +47,20 @@ public override IResourceHandler GetResourceHandler(IWebBrowser browserControl,
         /// requests starting with the given base url.</param>
         public void RegisterProvider(string baseurl, IResourceProvider provider)
         {
-            supportedSchemes.Add(provider.Scheme);
-            resourceProviders.Add(baseurl, provider);
+            if (!resourceProviders.ContainsKey(baseurl))
```

Comment:
I think we should not be changing resource provider for the same url. Url should have additional data to differentiate the different scenarios.